packages/sales-api-service/src/services/__tests__/contacts.service.spec.js

```diff
@@ -27,33 +28,55 @@ describe('contacts service', () => {
 
     it('resolves an existing contact by id', async () => {
       const mockPayload = mockContactWithIdPayload()
-      const contact = await resolveContactPayload(mockPayload)
+      const mockPermission = mockPermissionPayload()
+      const contact = await resolveContactPayload(mockPermission, mockPayload)
       expect(contact.isNew()).toBeFalsy()
       expect(dynamicsLib.findById).toHaveBeenCalledWith(Contact, mockPayload.id)
     })
 
     it('creates a new contact entity if no contact found for an id', async () => {
       const mockPayload = mockContactWithIdPayload()
+      const mockPermission = mockPermissionPayload()
       dynamicsLib.findById.mockImplementationOnce(() => undefined)
-      const contact = await resolveContactPayload(mockPayload)
+      const contact = await resolveContactPayload(mockPermission, mockPayload)
       expect(contact.isNew()).toBeTruthy()
       expect(dynamicsLib.findById).toHaveBeenCalledWith(Contact, mockPayload.id)
     })
 
     it('resolves an existing contact by key fields', async () => {
       const mockPayload = mockContactPayload()
-      const contact = await resolveContactPayload(mockPayload)
+      const mockPermission = mockPermissionPayload()
+      const contact = await resolveContactPayload(mockPermission, mockPayload)
       expect(contact.isNew()).toBeFalsy()
       expect(dynamicsLib.findByExample).toHaveBeenCalledWith(findByExampleCallExpectation)
     })
 
     it('creates a new contact entity if no contact found key fields', async () => {
       const mockPayload = mockContactPayload()
+      const mockPermission = mockPermissionPayload()
       dynamicsLib.findByExample.mockImplementationOnce(() => [])
-      const contact = await resolveContactPayload(mockPayload)
+      const contact = await resolveContactPayload(mockPermission, mockPayload)
       expect(contact.isNew()).toBeTruthy()
       expect(dynamicsLib.findByExample).toHaveBeenCalledWith(findByExampleCallExpectation)
     })
+
+    it('licence is not 12 months and contact exists in CRM', async () => {
+      const mockPayload = mockContactPayload()
+      const mockPermission = {
+        licenceLength: '1D'
+      }
+      const contact = await resolveContactPayload(mockPermission, mockPayload)
+      expect(contact.preferredMethodOfConfirmation.description).not.toEqual(contact.shortTermPreferredMethodOfConfirmation.description)
+    })
+
+    it('licence is 12 months and contact exists in CRM', async () => {
+      const mockPayload = mockContactPayload()
+      const mockPermission = {
+        licenceLength: '12M'
+      }
+      const contact = await resolveContactPayload(mockPermission, mockPayload)
+      expect(contact.preferredMethodOfConfirmation.description).toEqual(contact.shortTermPreferredMethodOfConfirmation.description)
+    })
```

Comment:
You have several situations, I'm not sure you're covering all of them:
1) Contact exists in CRM and is buying a short term licence. Their short term confirmation preference should be updated _without changing the long term confirmation preference_
2) Contact exists in CRM and is buying a long term licence. Their long term confirmation preference should be updated _without changing the short term confirmation preference_
3) Contact doesn't exist in CRM and is buying a short term licence. Their short term confirmation preference should be updated
4) Contact doesn't exist in CRM and is buying a long term licence. Their long term confirmation preference should be updated

The additional clause introduced in the first two cases may require four tests (so six in total). An ideal use case for `it.each`, I'd say.